source/data-modeling/persistence-configuration.txt

```diff
@@ -0,0 +1,296 @@
+
+The preceding code creates a new inflection in the current instance of the
+``ActiveSupport::Inflector`` class. As a result, {+odm+} stores
+``Alumnus`` model class documents in the ``alumni`` collection.
+
+To learn more about inflection methods, see the documentation for
+`Active Support Inflections <https://api.rubyonrails.org/classes/ActiveSupport/Inflector/Inflections.html>`__.
+
+.. note:: BSON Document Structure
+
+   When {+odm+} stores a document in a database, it serializes the Ruby object
+   to a BSON document that has the following structure:
+
+   .. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+      :language: ruby
+      :start-after: start BSON model
+      :end-before: end BSON model
+
+Persistence Context Attributes
+------------------------------
+
+Every model class contains the following methods, which you can use to retrieve
+information about where {+odm+} persists the model:
+
+- ``client_name``: Retrieves the client name
+- ``database_name``: Retrieves the database name
+- ``collection_name``: Retrieves the collection name
+
+The following example shows how to retrieve and print the names of the client, 
+database, and collection where documents for the ``Band`` class are persisted:
+
+.. io-code-block::
+
+  .. input:: /includes/data-modeling/persistence-configuration.rb
+     :language: ruby
+     :start-after: start persistence context attributes
+     :end-before: end persistence context attributes
+
+  .. output::
+     :language: ruby
+     :visible: false
+
+     default
+
+     my_bands
+
+     bands
+
+Customize Your Persistence Configuration
+----------------------------------------
+
+{+odm+} provides both model-level and runtime options for customizing your 
+persistence configuration. The following sections describe these options.
+
+Model-Level Persistence Options
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Perhaps you want to store your model's documents in a collection with a 
+different name than the pluralized form of the model class name.
+
+You can use the ``store_in`` macro to change the collection, database, or client
+where {+odm+} stores a model's documents. The following example shows how
+to use the ``store_in`` macro to store documents from the ``Person`` class in
+a collection called ``citizens`` in the ``other`` database within a client
+named ``analytics``.
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+   :language: ruby
+   :start-after: start store_in example
+   :end-before: end store_in example
+
+The ``store_in`` macro can also accept a lambda. For example, you might want to 
+use this pattern in a multi-tenant application,
+where multiple users share common access to an application. You can use a lambda
+to perform operations on different databases based on the current thread so that
+users cannot access each others' data. The following example stores documents in
+a database determined by the current thread:
```

Comment:
S: when you introduce a new thing, it can be helpful to go general -> specific:

1. Here's this lambda pattern
2. Here's why it's cool
3. Here's when you might want to use it and why
4. Code example

something like this:

```suggestion
The ``store_in`` macro can also accept a lambda. Lambdas let you
[do what cool stuff, like thread restrictions?].

You might want to use this pattern in a multi-tenant application,
where multiple users share common access to an application. By using
a lambda, you can prevent users from accessing each others' data.

The following example stores documents in
a database determined by the current thread:
```